microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+
+However, sometimes you may need to focus on specific parts of the data that the **Memory** tool doesn't show. Fortunately DevTools makes it possible to export the entire memory data as a `heapsnapshot` file which you can then analyze and visualize with your own tools.
+
+This article describes the structure and content of `heapsnapshot` files so you can build your own analysis tools.
+
+
+<!-- ====================================================================== -->
+## Record a heap snapshot
+
+To export a `heapsnapshot` file, you first need to record it in the **Memory** tool.
+
+1. In Microsoft Edge, navigate to the website you want to export the data from.
+
+1. Press `Ctrl`+`Shift`+`I` (Windows, Linux) or `Command`+`Option`+`I` (macOS) to open Devtools.
+
+1. Open the **Memory** tool.
+
+1. Select **Heap snapshot** and then click **Take snapshot**.
+
+To find out more see [Record heap snapshots using the Memory tool](heap-snapshots.md).
+
+
+<!-- ====================================================================== -->
+## Export and view a `heapsnapshot` file
+
+Once you have recorded a heap snapshot, you can export it.
+
+1. In the **Memory** tool left sidebar, click **Save** next to the heap snapshot item you just recorded.
+
+1. Change the file extension from `.heapsnapshot` to `.json` to make it easier to open the file in a text editor.
+
+1. Open the saved file in a text editor such as Visual Studio Code.
+
+1. To facilitate reading the `heapsnapshot` file, re-format the JSON: in Visual Studio Code, right-click on the code and choose **Format document**.
+
+Note that the resulting file will be different every time you record and export a heap snapshot. Heap snapshots are dynamically generated based on the content of the web application currently being inspecting in DevTools.
+
+
+<!-- ====================================================================== -->
+## Overview of the `heapsnapshot` file format
+
+The memory used by a web application is organized in a graph by V8, the JavaScript engine used by Microsoft Edge. A graph is a data type that is composed of nodes (points on the graph) and edges (links between the points).
+
+The data in the `heapsnapshot` file represents that graph efficiently and in a way that facilitates the transfer of data as chunks between the browser process and DevTools. The `heapsnapshot` file contains a flattened representation of the relations between nodes and edges as a JSON object containing arrays of numbers and strings. Even if the file originally has a `.heapsnapshot` extension, it contains JSON.
+
+The data has two main parts:
+
+* The metadata, which contains all of the information you need in order to parse the arrays of data found in the payload to re-create the graph.
+* The payload, which contains the actual data needed to re-create the graph.
```

Comment:
```suggestion
* The payload, which contains the actual data that's needed to re-create the graph.
```

Reply:
finding: American Heritage dict. supports both recreate and re-create, I'd write re-create.

https://www.ahdictionary.com/word/search.html?q=recreate